File: ServeurWebSocket/final/amin nchalh lekher/victorious-master/src/Entity/Team.php
<?php

namespace App\Entity;

use DateTimeImmutable;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Entity\File as EmbeddedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * Team
 *
 * @ORM\Table(name="team")
 * @ORM\Entity
 * @Vich\Uploadable
 */
 class Team
         {
             /**
              * @var int
              *
              * @ORM\Column(name="Id_Team", type="integer", nullable=false)
              * @ORM\Id
              * @ORM\GeneratedValue(strategy="IDENTITY")
              */
             private $id;
         
             /**
              * @var string
              *
              * @Assert\NotBlank
              * @ORM\Column(name="Team_Name", type="string", length=20, nullable=false)
              */
             private $teamName;
         
             /**
              * @var int
              * @Assert\NotBlank
              * @ORM\Column(name="Nb_Players", type="integer", nullable=false)
              */
             private $nbPlayers;
         
              
         
             /**
              * @var string
              *@ORM\ManyToMany(targetEntity="Player", inversedBy="teams")
              * @ORM\Column(name="Players", type="string", length=50, nullable=false)
              */
             private $players;
         
             /**
              * @var string
              *
              * @ORM\Column(name="Favorite_Games", type="string", length=100, nullable=false)
              */
             private $favoriteGames;
         
             /**
              * @var string
              *
              * @ORM\Column(name="Team_Desciption", type="string", length=500, nullable=false)
              */
             private $teamDesciption;
         
             /**
              * @var string
              *
              * @ORM\Column(name="Password", type="string", length=30, nullable=false)
              */
             private $password;
         
             /**
              * @var string
              *
              * @ORM\Column(name="Team_Mail", type="string", length=30, nullable=false)
              */
             private $teamMail;
         
             /**
              * @var string|null
              *
              * @ORM\Column(name="logo", type="text", length=16777215, nullable=true)
              */
             private $logo;
               /**
              * NOTE: This is not a mapped field of entity metadata, just a simple property.
              * 
              * @Vich\UploadableField(mapping="team_images", fileNameProperty="image")
              * 
              * @var File|null
              */
             private $imageFile;
      
             /**
              * @ORM\Column(name="tournaments_won",type="integer")
              */
             private $nb;
         
             public function getIdTeam(): ?int
             {
                 return $this->id;
             }
         
             public function getTeamName(): ?string
             {
                 return $this->teamName;
             }
         
             public function setTeamName(string $teamName): self
             {
                 $this->teamName = $teamName;
         
                 return $this;
             }
         
             public function getNbPlayers(): ?int
             {
                 return $this->nbPlayers;
             }
         
             public function setNbPlayers(int $nbPlayers): self
             {
                 $this->nbPlayers = $nbPlayers ;
         
                 return $this;
             }
            
             public function getPlayers(): ?string
             {
                 return $this->players;
             }
         
             public function setPlayers(string $players): self
             {
                 $this->players = $players;
         
                 return $this;
             }
         
             public function getFavoriteGames(): ?string
             {
                 return $this->favoriteGames;
             }
         
             public function setFavoriteGames(string $favoriteGames): self
             {
                 $this->favoriteGames = $favoriteGames;
         
                 return $this;
             }
         
             public function getTeamDesciption(): ?string
             {
                 return $this->teamDesciption;
             }
         
             public function setTeamDesciption(string $teamDesciption): self
             {
                 $this->teamDesciption = $teamDesciption;
         
                 return $this;
             }
         
             public function getPassword(): ?string
             {
                 return $this->password;
             }
         
             public function setPassword(string $password): self
             {
                 $this->password = $password;
         
                 return $this;
             }
         
             public function getTeamMail(): ?string
             {
                 return $this->teamMail;
             }
         
             public function setTeamMail(string $teamMail): self
             {
                 $this->teamMail = $teamMail;
         
                 return $this;
             }
         
             public function getLogo(): ?string
             {
                 return $this->logo;
             }
         
             public function setLogo(?string $logo): self
             {
                 $this->logo = $logo;
         
                 return $this;
             }
             public function addPlayer(Player $players): self
             {
                 if (!$this->players->contains($players)) {
                     $this->players[] = $players;
                     $players->setTeam($this);
                 }
         
                 return $this;
             }
             public function __toString() {
                 return $this->teamName;
             }
             
             public function setImageFile(?File $logo = null)
             {
                 $this->imageFile = $logo;
         
                 if (null !== $logo) {
                     // It is required that at least one field changes if you are using doctrine
                     // otherwise the event listeners won't be called and the file is lost
                     $this->updatedAt = new \DateTimeImmutable();
                 }
             }
         
             public function getImageFile()
             {
                 return $this->imageFile;
             }
         
             public function setImage(?string $logo): self
             {
                 $this->logo = $logo;
                 return $this;
             }
         
             public function getImage(): ?string
             {
                 return $this->logo;
             }
   
             public function getNb(): ?int
             {
                 return $this->nb;
             }

             public function setNb(int $nb): self
             {
                 $this->nb = $nb;

                 return $this;
             }
         }

File: ServeurWebSocket/final/amin nchalh lekher/victorious-master/Victorious_Mobile/src/entitie/Team.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entitie;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Objects;

/**
 *
 * @author jasser
 */
public class Team {
    private int Id_Team;
     String Team_Name;
    private int Nb_Players;
    private String Players;
    private String Favorite_Games;
    private String Team_Desciption ;
    private String Password ;
    private String Team_Mail;
    private String logo;

    public Team() {
    }

    public Team(String Team_Name, int Nb_Players, String Players, String Favorite_Games, String Team_Desciption, String Password, String Team_Mail, String logo) {
        this.Team_Name = Team_Name;
        this.Nb_Players = Nb_Players;
        this.Players = Players;
        this.Favorite_Games = Favorite_Games;
        this.Team_Desciption = Team_Desciption;
        this.Password = Password;
        this.Team_Mail = Team_Mail;
        this.logo = logo;
    }

    public Team(String Team_Name, String Players, String Favorite_Games, String Team_Desciption, String Password, String Team_Mail, String logo) {
        this.Team_Name = Team_Name;
        this.Players = Players;
        this.Favorite_Games = Favorite_Games;
        this.Team_Desciption = Team_Desciption;
        this.Password = Password;
        this.Team_Mail = Team_Mail;
        this.logo = logo;
    }

    public Team(int Id_Team, String Team_Name, int Nb_Players,String Players,String Favorite_Games, String Team_Desciption, String Password,String logo) {
        this.Id_Team = Id_Team;
        this.Team_Name = Team_Name;
        this.Nb_Players = Nb_Players;
        this.Players = Players;
        this.Favorite_Games = Favorite_Games;
        this.Team_Desciption = Team_Desciption;
        this.Password = Password;
        this.Team_Mail = Team_Mail;
        this.logo=logo;
    }

    public int getId_Team() {
        return Id_Team;
    }

    public String getLogo() {
        return logo;
    }

    public void setLogo(String logo) {
        this.logo = logo;
    }

    public void setId_Team(int Id_Team) {
        this.Id_Team = Id_Team;
    }

    public String getTeam_Name() {
        return Team_Name;
    }

    public void setTeam_Name(String Team_Name) {
        this.Team_Name = Team_Name;
    }

    public int getNb_Players() {
        return Nb_Players;
    }

    public void setNb_Player(int Nb_Players) {
        this.Nb_Players = Nb_Players;
    }

    public String getPlayers() {
        return Players;
    }

    public void setPlayers(String Players) {
        this.Players = Players;
    }

    public String getFavorite_Games() {
        return Favorite_Games;
    }

    public void setFavorite_Games(String Favorite_Games) {
        this.Favorite_Games = Favorite_Games;
    }

    public String getTeam_Desciption() {
        return Team_Desciption;
    }

    public void setTeam_Desciption(String Team_Desciption) {
        this.Team_Desciption = Team_Desciption;
    }

    public String getPassword() {
        return Password;
    }

    public void setPassword(String Password) {
        this.Password = Password;
    }

    public String getTeam_Mail() {
        return Team_Mail;
    }

    public void setTeam_Mail(String Team_Mail) {
        this.Team_Mail = Team_Mail;
    }

    @Override
    public String toString() {
        return "Team{" + "Id_Team=" + Id_Team + ", Team_Name=" + Team_Name + ", Nb_Player=" + Nb_Players + ", Players=" + Players + ", Favorite_Games=" + Favorite_Games + ", Team_Desciption=" + Team_Desciption + ", Password=" + Password + ", Team_Mail=" + Team_Mail + '}';
    }

    @Override
    public int hashCode() {
        int hash = 3;
        hash = 19 * hash + this.Id_Team;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Team other = (Team) obj;
        if (this.Id_Team != other.Id_Team) {
            return false;
        }
        if (this.Nb_Players != other.Nb_Players) {
            return false;
        }
        if (!Objects.equals(this.Team_Name, other.Team_Name)) {
            return false;
        }
        if (!Objects.equals(this.Favorite_Games, other.Favorite_Games)) {
            return false;
        }
        if (!Objects.equals(this.Team_Desciption, other.Team_Desciption)) {
            return false;
        }
        if (!Objects.equals(this.Password, other.Password)) {
            return false;
        }
        if (!Objects.equals(this.Team_Mail, other.Team_Mail)) {
            return false;
        }
        if (!Objects.equals(this.Players, other.Players)) {
            return false;
        }
        return true;
    }

    

  
    
}
